clients/mediaconvert.go

```diff
@@ -90,7 +91,7 @@ func (mc *MediaConvert) Transcode(ctx context.Context, args TranscodeJobArgs) er
 	if path.Base(args.HLSOutputFile.Path) != "index.m3u8" {
 		return fmt.Errorf("target URL must be an `index.m3u8` file, found %s", args.HLSOutputFile)
 	}
-	targetDir := getTargetDir(args.HLSOutputFile)
+	targetDir := path.Join(config.RandomTrailer(10), getTargetDir(args.HLSOutputFile))
```

Comment:
AFAIK this will generate a completely random string, not use the request ID. Could we reuse `args.RequestID` here instead?

On another note, I think it would be much easier if we add the request ID to the end of this path, not to the beginning. So when we're looking for specific input/output files we can start from the playback ID of the asset and then find the corresponding request IDs inside. If we start with the request ID (or worse, with this random ID here) it will have an additional layer of indirection where we'll need to go through the logs to find out where each file went.

On a final and less important note, I'd also move this logic to the `getTargetDir` function, as the sole purpose of it is to generate this directory. It should have the logic of adding any randomness to it as well.

Reply:
Oh just saw Thom's comment about not wanting to use the request ID here. I guess it makes sense, but it's yet another reason not to organize all the files on a folder with this random string instead, will get nearly impossible to find later (not sure if we're even logging this anywhere).

Personally, these files are temporary and so we don't have any restriction on their format or permanence after the output files are copied back to their final location. It seems fine to me to use the request ID in that case and think of something else if we ever change them to be "file name incompatible". We could perhaps URL-encode the request ID here just in case.

WDYT @thomshutt? If you still think it's better not to use them in these temporary file names, then I'd advocate strongly for my first point, of not using them as the root folder but as the final folder in the path.

Reply:
To your points:

1. About using `requestID` vs `randomTrailer()` => I'm ok with both. Request ID is simpler for tracking, but on the other hand it may make the assumption we won't change it, so I get the point from Thom of not using it. @thomshutt please comment on this one.

2. Move logic to `getTargetDir` => ok, I'll move it.

3. Random string at the end of the path => I think it's a good idea. I'll move it.

Reply:
I think that if there's no other easy way for us to correlate a given transcode to where the file is stored, then I'm okay with us using RequestID here - sorry for all the back and forth here @leszko!

Reply:
> 3. Random string at the end of the path => I think it's a good idea. I'll move it.

Actually I just realized it's not that easy to do, because we copy the whole path recursively. Keeping it at the beginning.

Reply:
> Actually I just realized it's not that easy to do, because we copy the whole path recursively. Keeping it at the beginning.

Hmm, not sure if I understand. Conceptually, if we had a logic like:
 - `path.Join(args.RequestID, getTargetDir(args.HLSOutputFile))
why can't we make it
 - `path.Join(getTargetDir(args.HLSOutputFile), args.RequestID)`
?

Doesn't really matter if the inner `getTargetDir` is recursive, only matters where we add the request ID in the output of that process. Am I misunderstanding something?

Reply:
Just to be clear why I'm insisting on this, I think this debug experience would be pretty neat:
 - Get the task/asset ID from the corresponding asset to investigate, from Studio API
 - From either of those, find the asset playback ID
 - With that, access the S3 transfer bucket on the path `output/hls/<playbackId>`
 - Now here we'll find 1 bucket for each attempt of transcoding the file. Since Studio does 3 retries we'll find 3 folders, each with the corresponding request ID
 - Dig into each request ID ✨ 
 
That in opposition to first having to find all the request IDs for a given playback ID in the logs, then inverting the search in S3 starting with the request ID instead (which their dashboard also makes it harder to do).

Reply:
@victorges I think you're right. Updated in https://github.com/livepeer/catalyst-api/pull/355. Please approve.